src/libraries/System.Memory/tests/Span/CommonPrefixLength.T.cs

```diff
@@ -0,0 +1,124 @@
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length + 1], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length], new char[length + 1], NonDefaultEqualityComparer<char>.Instance));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(15)]
+        public static void SecondShorterAllEqual_ReturnsSecondLength(int length)
+        {
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length]));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length], null));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<char>)new char[length + 1], new char[length], NonDefaultEqualityComparer<char>.Instance));
+
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length]));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length], null));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length], EqualityComparer<char>.Default));
+            Assert.Equal(length, MemoryExtensions.CommonPrefixLength((Span<char>)new char[length + 1], new char[length], NonDefaultEqualityComparer<char>.Instance));
+        }
+
+        [Fact]
+        public static void PartialEquals_ReturnsPrefixLength_ValueType()
+        {
+            byte[] arr1 = new byte[] { 1, 2, 3, 4, 5 };
+            byte[] arr2 = new byte[] { 1, 2, 3, 6, 7 };
+
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<byte>)arr1, arr2));
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<byte>)arr1, arr2, null));
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<byte>)arr1, arr2, EqualityComparer<byte>.Default));
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((ReadOnlySpan<byte>)arr1, arr2, NonDefaultEqualityComparer<byte>.Instance));
+
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((Span<byte>)arr1, arr2));
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((Span<byte>)arr1, arr2, null));
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((Span<byte>)arr1, arr2, EqualityComparer<byte>.Default));
+            Assert.Equal(3, MemoryExtensions.CommonPrefixLength((Span<byte>)arr1, arr2, NonDefaultEqualityComparer<byte>.Instance));
+        }
+
+        [Fact]
+        public static void PartialEquals_ReturnsPrefixLength_ReferenceType()
```

Comment:
These tests won't catch the introduction of a bug where we inadvertently ignore the provided comparer. I recommend also including a test that passes `StringComparer.OrdinalIgnoreCase` and validating that it performs the correct comparison against inputs.